moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -796,12 +795,10 @@ bool ServoCalcs::enforceSRDFPositionLimits()
 // Is handled differently for position vs. velocity control.
 void ServoCalcs::suddenHalt(trajectory_msgs::JointTrajectory& joint_trajectory)
 {
-  if (joint_trajectory.points.empty())
-  {
-    joint_trajectory.points.push_back(trajectory_msgs::JointTrajectoryPoint());
-    joint_trajectory.points[0].positions.resize(num_joints_);
-    joint_trajectory.points[0].velocities.resize(num_joints_);
-  }
+  joint_trajectory.points.clear();
+  joint_trajectory.points.push_back(trajectory_msgs::JointTrajectoryPoint());
+  joint_trajectory.points[0].positions.resize(num_joints_);
+  joint_trajectory.points[0].velocities.resize(num_joints_);
 
   for (std::size_t i = 0; i < num_joints_; ++i)
   {
```

Comment:
in this block, do you want to change [i] to .at(i)?  Would have made this a lot easier to track down

Reply:
Yes, and I was thinking yesterday we should try to use `at(i)` by default wherever we can to avoid these sorts of issues in the future.  The runtime cost is not that bad if the array size is small.  If the array is large it will prevent vectorization.

Reply:
The best part of at(i) is that it will throw an exception here if we try to access past the end of the array.

Reply:
That's the whole point of `at()`, yes. I'm undecided on whether to use `at()` or `operator[]`.
I would be surprised to see the former in MoveIt's most real-time capable module, but it's probably hardly measurable with our use-cases.

Reply:
Agree with @v4hn, would be good to have an idea of the cost. Remember someone might want this to run at 500-1000 Hz or more

Reply:
Replacing it right here is 100% fine because it should only get called rarely. Not part of the main control loop.

I tested this out for the heck of it. On the master branch, the control loop rate varies from about 4.95 to 10.1 kHz. Occasional spikes as slow as 98.4 Hz.  :O

Blindly replacing [i] with .at(i) everywhere doesn't change much. 3.9 kHz - 11.29 kHz, with occasional spikes as slow as 98.6 Hz

Reply:
If you can add instructions to benchmark this (or maybe even automate it), [here](https://github.com/ros-planning/moveit/tree/master/moveit_ros/planning/planning_components_tools/src) would be a great place to put your script.

Reply:
If I remove the switch to `at()` and the other unrelated refactor of the initialization of the other fields of `joint_trajectory` can we accept this change and revisit those in a future issue?  I would like to get the bug fix accepted and then move on the less trivial discussions around tradeoffs in implementation separately.